python-avd/pyavd/_eos_designs/structured_config/__init__.py

```diff
@@ -57,21 +60,27 @@
 
 
 def get_structured_config(
-    vars: dict,  # noqa: A002
+    *,
+    hostname: str,
+    hostvars: Mapping,
     input_schema_tools: AvdSchemaTools,
+    all_facts: Mapping[str, EosDesignsFacts],
     result: dict,
     templar: object | None = None,
-    *,
     validate: bool = True,
-) -> dict:
+) -> EosCliConfigGen | None:
     """
     Generate structured_config for a device.
 
     Args:
-        vars:
-            The variable for the device
+        hostname:
+            The hostname of the device
+        hostvars:
+            The variables for the device
         input_schema_tools:
             An AvdSchemaTools object used to validate the input variables if enabled.
+        all_facts:
+            Map of all devices and their facts.
```

Comment:
need to update docstring return below (can't comment there)

Reply:
done